odoo/addons/base/models/ir_actions.py

```diff
@@ -403,6 +403,8 @@ def _select_objects(self):
                                     help="Provide the field used to link the newly created record "
                                          "on the record on used by the server action.")
     fields_lines = fields.One2many('ir.server.object.lines', 'server_id', string='Value Mapping', copy=True)
+    groups_id = fields.Many2many('res.groups', 'ir_act_server_group_rel',
+                                 'act_id', 'gid', string='Groups')
```

Comment:
Can you add a check on `groups_id` in the method `run()`, so that an action with at least one group cannot be run by a user without any of the groups (unless it is the superuser) ?  That would make the new feature effective when trying to execute the action.